aws/resources/ec2.go

```diff
@@ -140,722 +130,34 @@ func (ei *EC2Instances) nukeAll(instanceIds []*string) error {
 	logging.Debugf("Terminating all EC2 instances in region %s", ei.Region)
 
 	params := &ec2.TerminateInstancesInput{
-		InstanceIds: instanceIds,
+		InstanceIds: aws.ToStringSlice(instanceIds),
 	}
 
-	_, err = ei.Client.TerminateInstancesWithContext(ei.Context, params)
+	_, err = ei.Client.TerminateInstances(ei.Context, params)
 	if err != nil {
 		logging.Debugf("[Failed] %s", err)
 		return errors.WithStackTrace(err)
 	}
 
-	err = ei.Client.WaitUntilInstanceTerminatedWithContext(ei.Context, &ec2.DescribeInstancesInput{
-		Filters: []*ec2.Filter{
+	waiter := ec2.NewInstanceTerminatedWaiter(ei.Client)
+	err = waiter.Wait(ei.Context, &ec2.DescribeInstancesInput{
```

Comment:
The tests fail at this line, and I'm not sure what it's complaining about. Ideas?

<details>
<summary>Details</summary>

```
--- FAIL: TestEc2InstancesWithEIP_NukeAll (0.00s)
panic: cannot create context from nil parent [recovered]
	panic: cannot create context from nil parent

goroutine 125 [running]:
testing.tRunner.func1.2({0x10375f7c0, 0x103ba6620})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1632 +0x1bc
testing.tRunner.func1()
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1635 +0x334
panic({0x10375f7c0?, 0x103ba6620?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/runtime/panic.go:785 +0x124
context.WithDeadlineCause({0x0?, 0x0?}, {0x1041e9e20?, 0xd18c2e2800?, 0x1041e9e20?}, {0x0?, 0x0?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/context/context.go:620 +0x368
context.WithDeadline(...)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/context/context.go:612
context.WithTimeout({0x0, 0x0}, 0xd18c2e2800)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/context/context.go:690 +0x50
github.com/aws/aws-sdk-go-v2/service/ec2.(*InstanceTerminatedWaiter).WaitForOutput(0x14000870420, {0x0, 0x0}, 0x14000828280, 0xd18c2e2800, {0x0, 0x0, 0xb0?})
	/my/path/to/go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/ec2@v1.193.0/api_op_DescribeInstances.go:1428 +0x17c
github.com/aws/aws-sdk-go-v2/service/ec2.(*InstanceTerminatedWaiter).Wait(...)
	/my/path/to/go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/ec2@v1.193.0/api_op_DescribeInstances.go:1402
github.com/gruntwork-io/cloud-nuke/aws/resources.(*EC2Instances).nukeAll(0x14000a43dd8, {0x14000a43e98, 0x1, 0x102ea78e0?})
	/my/path/to/cloud-nuke/aws/resources/ec2.go:143 +0x454
github.com/gruntwork-io/cloud-nuke/aws/resources.TestEc2InstancesWithEIP_NukeAll(0x14000449040)
	/my/path/to/cloud-nuke/aws/resources/ec2_test.go:177 +0x178
testing.tRunner(0x14000449040, 0x103ba4398)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1690 +0xe4
created by testing.(*T).Run in goroutine 1
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1743 +0x314
```

</details>

Reply:
Did you try to provide `context` when creating the `mock Client` in your test? e.q.

```go
ei := EC2Instances{
		BaseAwsResource: BaseAwsResource{
			Context: context.Background(),
		},
		Client: mockedEC2Instances{
		...
```

Reply:
Yes, this worked! Thank you @wakeful